app/src/main/java/org/oppia/app/shim/IntentFactoryShim.kt

```diff
@@ -0,0 +1,39 @@
+package org.oppia.app.shim
+
+import android.content.Context
+import android.content.Intent
+import androidx.fragment.app.FragmentActivity
+import org.oppia.app.drawer.KEY_NAVIGATION_PROFILE_ID
+import org.oppia.app.home.recentlyplayed.RecentlyPlayedActivity
+import org.oppia.app.profile.ProfileActivity
+import org.oppia.app.topic.TopicActivity
+import javax.inject.Inject
+
+class IntentFactoryShim @Inject constructor() : IntentFactoryShimInterface {
```

Comment:
Please make sure all public classes, enums, interfaces, and other members have documentation: https://developer.android.com/kotlin/style-guide#usage. Setters & getters often don't need KDocs, but only if they are trivial. Some of the new setters & getters you're introducing to replace existing adapters aren't trivial--please add documentation for those, as well.

Reply:
Okay - will do!